File: sdk/flutter/tag_data_translation/ios/Classes/TagDataTranslationPlugin.swift
import Flutter

/// Flutter plugin registration for TagDataTranslation.
///
/// The actual translation is handled via dart:ffi calling the NativeAOT
/// static library directly. This class only exists to satisfy Flutter's
/// plugin registration requirements.
public class TagDataTranslationPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // no-op: dart:ffi handles all native calls directly
    }
}

File: sdk/flutter/tag_data_translation/android/src/main/kotlin/nl/mimasu/tdt/TagDataTranslationPlugin.kt
package nl.mimasu.tdt

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Flutter plugin registration for TagDataTranslation.
 *
 * The actual translation is handled via dart:ffi calling the NativeAOT
 * shared library directly. This class only exists to satisfy Flutter's
 * plugin registration requirements.
 */
class TagDataTranslationPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // no-op: dart:ffi handles all native calls directly
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // no-op
    }
}
